src/guessing_game_with_struct.rs

```diff
@@ -0,0 +1,55 @@
+use std::io;
+use rand::Rng;
+use std::cmp::Ordering;
+
+#[derive(Debug)]
+struct Game {
+    secret_number: u32,
+    history: Vec<u32>,
+}
+
+impl Game {
+    fn build() -> Self {
+        Game {
+            secret_number: rand::thread_rng().gen_range(1..10),
+            history: Vec::new()
+        }
+    }
+
+    fn ask_for_number(&self) -> Result<u32, &str> {
+        let mut my_guess = String::new();
+        io::stdin().read_line(&mut my_guess).expect("my input was wrond");
+        match my_guess.trim().parse() {
+            Ok(num) => Ok(num),
+            Err(_) => Err("error"),
+        }
+    }
+
+    fn compare_numbers(&self, my_guess: &u32) -> Result<bool, &str> {
+        match self.secret_number.cmp(&my_guess) {
+            Ordering::Less => { Err("lower") } 
+            Ordering::Greater => { Err("higher") } 
+            Ordering::Equal => Ok(true)
+        }
+    }
+}
+
```

Comment:
Entry point for the guessing game. 
Create a new instance and start the loop with asking for a number and compare